File: Tutorial/Examples/Publisher_subscriber/Producer_C#/Publisher.cs
using System;
using System.Collections.Generic; 
using System.Threading.Tasks;
using System.Linq;
using System.Text;
using RabbitMQ.Client;

// Create connection.
var factory = new ConnectionFactory{ HostName = "localhost"};

// Instantiate the connection and one channel.
using var connection = factory.CreateConnection();
using var channel = connection.CreateModel();
channel.ExchangeDeclare(exchange: "pubsub", type: ExchangeType.Fanout);

// Instantiate a random number generator
Random random = new Random();
// Variable setup for message construction.
int x = random.Next(1, 6);
int y = random.Next(1, 6);
int z = random.Next(1, 6);
int d = random.Next(2, 5);


// Declare message to produce and encode it as bytes
var message = string.Format("{0},{1},{2},{3}", x, y, z, d);
var encodedMessage = Encoding.UTF8.GetBytes(message);


// Publish to the pubsub exchange.
channel.BasicPublish(exchange: "pubsub", "", null, encodedMessage);
Console.WriteLine($"Sent message: {message}");

File: Tutorial/Examples/Publisher_subscriber/Producer_Python/Publisher.py
import pika
from pika.exchange_type import ExchangeType
import random

# Set connection parameters. If connecting to a real server localhost should be replaced by the server´s address.
connection_parameters = pika.ConnectionParameters('localhost')

# Instantiate a connection unsing the connection_parameters previously defined.
connection = pika.BlockingConnection(connection_parameters)

# Instantiate a channel.
channel = connection.channel()

# Instantiate an exchange.
channel.exchange_declare(exchange='pubsub', exchange_type=ExchangeType.fanout)

# Define the message to send.
x = random.randint(1, 6)
y = random.randint(1, 6)
z = random.randint(1, 6)
d = random.randint(2, 5)
message = f"{x},{y},{z},{d}"

# Publish the message.
channel.basic_publish(exchange='pubsub', routing_key='', body=message)

print(f"Sent message: {message}")

connection.close()
